### PR TITLE
Custom user permissions UI

### PR DESCRIPTION
**Pull request by itsrachelfish**
This PR updates the Lightning Node Connect page to include a new overlay which allows users to create macaroons with customized permissions for specific URIs as well as expiration dates.

## New option added to permissions dropdown
image

## Custom permissions overlay
image

## Custom session with an expiration date
image

Related to: #438
Depends on #450 & #457

### CODE REVIEW

File: app/src/components/common/v2/Text.tsx
Comment (jbrill) [inline app/src/components/common/v2/Text.tsx:35]: Should we stay consistent with
`color: ${props => (props.muted && 'props.theme.colors.gray')}`

File: app/src/components/common/v2/Text.tsx
Comment (jbrill) [inline app/src/components/common/v2/Text.tsx:36]: Same as above

**Comment by jbrill** [comment]
One future thing that could be helpful is to create a common component library as a node package for us to consume in both terminal and lit. This way we could avoid duplicated code, and maybe create a storybook for @dstrukt to view what's being used in the codebase

**Comment by jbrill** [comment]
Should `Liquidity Manager` and `Payments Manager` permission roles be in the dropdown as well?

**Comment by levmi** [comment]
> Should `Liquidity Manager` and `Payments Manager` permission roles be in the dropdown as well?

Design/product decision to just have those live in custom as well. Want to see if people use them or not. Will populate the dropdown with the most popular options based on feedback. Still unsure how popular those two will be, but it's our first stab at it.

**Comment by jbrill** [comment]
I can imagine the terminal lift will be significant for some of these custom permissions -- would those changes happen before the merge? Just curious

**Comment by jbrill** [comment]
The code looks very clean and can confirm the feature works very well 👍 

Will do a more thorough pass when it's ready for review!

File: app/src/components/common/FormDate.tsx
Comment (jbrill) [inline app/src/components/common/FormDate.tsx:62]: Do we want to use the `useCallback` format similar to what's used in Terminal?

File: app/src/components/connect/CustomSessionPage.tsx
Comment (jbrill) [inline app/src/components/connect/CustomSessionPage.tsx:98]: (reminder to tweak)

**Comment by dstrukt** [comment]
> One future thing that could be helpful is to create a common component library as a node package for us to consume in both terminal and lit. This way we could avoid duplicated code, and maybe create a storybook for @dstrukt to view what's being used in the codebase

This is interesting .. ACK for the shared library, and storybook if we keep it low effort export for time being.

File: app/src/store/views/addSessionView.ts
Comment (ellemouton) [inline app/src/store/views/addSessionView.ts:109]: these are the wrong way around :)

**Review by ellemouton — CHANGES_REQUESTED** [review]

**Comment by dstrukt** [comment]
Design Review:

- [ ] Add `24px` padding above Proxy Server input
Screen Shot 2022-11-28 at 12 10 16 PM

- [ ] Make the calendar icon our `offWhite` 
Screen Shot 2022-11-28 at 12 14 24 PM

- [ ] Make the calendar text copy click trigger the dropdown, only the icon does this (ideal, but not required)

- [ ] Conversely, the icon click on the select dropdown should trigger the dropdown too .. but this is an issue across LITD/TW, so another ideal, but not required, and might be worth a sep. issue, but defer to you

- [ ] Add a `200ms ease-in-out` transition to the background-color when hovering/selecting a Permission Type (just a nice to have)

**Review by dstrukt — CHANGES_REQUESTED** [review]
See above.

File: app/src/components/common/v2/FormSwitch.tsx
Comment (itsrachelfish) [reply app/src/components/common/v2/FormSwitch.tsx:17]: Yep :+1:

File: app/src/components/common/FormDate.tsx
Comment (itsrachelfish) [reply app/src/components/common/FormDate.tsx:62]: This was copied from the FormInput component. If we change it here we should probably update all of the input components to be consistent?

File: app/src/components/common/v2/Text.tsx
Comment (itsrachelfish) [reply app/src/components/common/v2/Text.tsx:35]: The difference is that the font weight always has a value. If the muted property doesn't exist there would be no color in the code you suggested.

**Comment by levmi** [comment]
That should be okay. Agree with Andy on the ideal behavior, but understand the tradeoff we made wrt usage of the browser controlled date picker and that's fine.

File: app/src/components/connect/CustomSessionPage.tsx
Comment (jamaljsr) [inline app/src/components/connect/CustomSessionPage.tsx:132]: The text labels shown in the UI should be extracted into the `en-US.json` file.

**Review by jamaljsr — APPROVED** [review]
tACK 🔥 Tested many different permission configurations and didn't run into any issues. 

Overall, the code looks great. I commented with a bunch of nits which aren't blocking this from being merged. Just some minor improvements to make it easy to grasp what's happening in the code.

File: app/src/components/connect/CustomSessionPage.tsx
Comment (dstrukt) [reply app/src/components/connect/CustomSessionPage.tsx:152]: @levmi Want to make sure there's visibility on the selection, do all of these options look good?

Comment (levmi) [reply app/src/components/connect/CustomSessionPage.tsx:152]: This is for expiry, correct? If so, LGTM. With custom, they can grant 1 day access, which I think people may want, but don't need to make a change to surface that at this stage given the existence of custom.

Comment (dstrukt) [reply app/src/components/connect/CustomSessionPage.tsx:152]: Yup, for expiry, and sounds good .. figured we'd adjust based on feedback, but wanted to make sure we were on the same page about initial MVP options.

**Review by dstrukt — APPROVED** [review]
LGTM!

Excellent work!

**Comment by itsrachelfish** [comment]
@dstrukt I added some code to automatically generate labels when none are defined. Users can still specify their own label when first viewing the add session option. However if no label is specified, the custom permissions page will automatically generate a label based on the permission type selected.

"My read-only session", "My liquidity session", "My loop session", etc.

**Comment by jamaljsr** [comment]
Thanks for the updates. The latest commit LGTM 🚀

**Review by levmi — APPROVED** [review]
All LGTM, great work!

**Comment by ellemouton** [comment]
needs a rebase :)

**Comment by itsrachelfish** [comment]
@levmi @ellemouton updated with the latest from master :+1:

**Review by ellemouton — APPROVED** [review]
tACK 🔥

**Comment by guggero** [comment]
@itsrachelfish do you want to rebase to get a clean commit history (all authored and committed by you) or should we just merge?

**Pull request merged**
